src/logcollector/read_mssql_log.c

```diff
@@ -28,13 +28,13 @@ void *read_mssql_log(logreader *lf, int *rc, int drop_it) {
     size_t str_len = 0;
     int need_clear = 0;
     char *p;
-    char str[OS_MAXSTR + 1];
-    char buffer[OS_MAXSTR + 1];
+    char str[OS_MAXSTR - OS_LOG_HEADER];
```

Comment:
Why are we substracting `OS_LOG_HEADER` here?

Reply:
Because the `__send_mssql_msg` function has a buffer inside it in which the buffer sent with a header will be concatenated. The maximum size of the header is defined by the tag.